instruqt-tracks/vault-advanced-data-protection-with-tokenization/track.yml

```diff
@@ -0,0 +1,376 @@
+
+    Once authenticated and authorized, Vault returns a token.
+    ```
+    //Login
+    secret, err1 := Vclient.Logical().Write(K8sAuthPath, config)
+    if err1 != nil {
+      log.Fatal(err)
+    }
+    Vclient.SetToken(secret.Auth.ClientToken)
+    ```
+    On line 62, we next pull database credentials from Vault in order to connect to PostgreSQL. In this setup, we are using dynamic secrets.
+    Dynamic Secrets are just-in-time credentials that are unique and short-lived for every application instance or pod.
+    Read more about dynamic secrets here: https://www.vaultproject.io/docs/secrets/databases/postgresql.
+    ```
+    //Pull dynamic database credentials
+    data, err := Vclient.Logical().Read("database/creds/vault_go_demo")
+    if err != nil {
+      log.Fatal(err)
+    }
+    username := data.Data["username"]
+    password := data.Data["password"]
+    ```
+
+    Next, open up the code editor tab called "Golang App Code - tokenization".
+
+    NOTE: Make sure to click the file name on the left to open up the editor.
+
+    On line 127 in the "AddRecord" function, we added the following line to tokenize the customer record's SSN before writing to Postgres.
+    ```
+    response, err := config.Vclient.Logical().Write("transform/encode/vault_go_demo", data)
+    if err != nil {
+      log.Fatal(err)
+    }
+    encval := response.Data["encoded_value"].(string)
+    ```
+    Last, we need to detokenize the SSN pulled from the database so we can view as plaintext.
+    Within the "Records" function (Line 58) we add the following:
+    ```
+    response, err := config.Vclient.Logical().Write("transform/decode/vault_go_demo", data)
+    if err != nil {
+      log.Fatal(err)
+    }
+    decval := response.Data["decoded_value"].(string)
+    ssn := decval
+    ```
+    With our code additions made, we can finally launch our updated application and test tokenization.
+  tabs:
+  - title: Terminal
+    type: terminal
+    hostname: kubernetes
+  - title: Golang App Code - init
+    type: code
+    hostname: kubernetes
+    path: /root/vault-go-demo-tokenization/config/db.go
+  - title: Golang App Code - tokenization
+    type: code
+    hostname: kubernetes
+    path: /root/vault-go-demo-tokenization/users/handlers.go
+  - title: Web App
+    type: service
+    hostname: kubernetes
+    path: /
+    port: 9090
+  - title: Vault-ui
+    type: service
+    hostname: kubernetes
+    path: /
+    port: 8200
+  difficulty: basic
+  timelimit: 10000
+- slug: launch-the-updated-app
+  id: xymeoyrbezvl
+  type: challenge
+  title: Launch the updated application
+  teaser: Deploy the updated app to securely tokenizen customer SSNs (PII data).
+  notes:
+  - type: text
+    contents: |-
+      We will now deploy our updated application to use following tokenization workflow.
+
+      ![Tokenization Workflow](https://github.com/Andrew-Klaas/instruqt-tokenization-lab/raw/main/assets/images/tokenization_workflow.png)
```

Comment:
Please get this from this repo or a new assets repo instead of from your personal repo.

Reply:
Fixed